### PR TITLE
[4.0] com_categories remove unused sidebar references

### PR DESCRIPTION
**Pull request by travisrisner**
Remove unused sidebar references for com_categories. I also removed a variable that didn't appear to be in use.

### CODE REVIEW

File: administrator/components/com_categories/Controller/DisplayController.php [hunk 1]
```diff
@@ -106,9 +106,6 @@ public function display($cachable = false, $urlparams = array())
 
 			// Push document object into the view.
 			$view->document = $document;
-
-			// Load the submenu.
-			CategoriesHelper::addSubmenu($model->getState('filter.extension'));
```
Comment (Quy) [inline administrator/components/com_categories/Controller/DisplayController.php:111]: Remove `use Joomla\Component\Categories\Administrator\Helper\CategoriesHelper;`.

File: administrator/components/com_categories/Helper/CategoriesHelper.php [hunk 2]
```diff
@@ -23,59 +23,6 @@
...
-		$component = $parts[0];
-
-		if (count($parts) > 1)
-		{
-			$section = $parts[1];
-		}
-
-		// Try to find the component helper.
-		$eName = str_replace('com_', '', $component);
-		$file = Path::clean(JPATH_ADMINISTRATOR . '/components/' . $component . '/helpers/' . $eName . '.php');
```
Comment (Quy) [inline administrator/components/com_categories/Helper/CategoriesHelper.php:53]: Remove `use Joomla\CMS\Filesystem\Path;`.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 7039795f73a1d1075d04697aa05e358c737ee3df<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25265">issues.joomla.org/tracker/joomla-cms/25265</a>.</sub>

**Comment by mothershedc** [comment]
I have tested :white_check_mark: this item successfully on 7039795

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25265">issues.joomla.org/tracker/joomla-cms/25265</a>.</sub>

**Comment by roland-d** [comment]
Thank you.

**Pull request merged**